modules/reindex/src/main/java/org/elasticsearch/index/reindex/remote/RemoteRequestBuilders.java

```diff
@@ -103,7 +103,9 @@ static String initialSearchPath(SearchRequest searchRequest) {
             searchRequest.source().storedField("_parent").storedField("_routing").storedField("_ttl");
             if (remoteVersion.before(Version.fromId(1000099))) {
                 // Versions before 1.0.0 don't support `"_source": true` so we have to ask for the _source in a funny way.
-                searchRequest.source().storedField("_source");
+                if (false == searchRequest.source().storedFields().fieldNames().contains("_source")) {
```

Comment:
Why don't you just do `if (!searchRequest.source().storedFields().fieldNames().contains("_source")) {` as contains returns a `boolean` ?